models/intermediate/int_quickbooks__cash_flow_classifications.sql

```diff
@@ -0,0 +1,97 @@
+       cast(account_type as {{ dbt.type_string() }}) as account_type,
+       cast(account_sub_type as {{ dbt.type_string() }}) as account_sub_type,
+       cast(account_number as {{ dbt.type_string() }}) as account_number,
+       cast(cash_flow_type as {{ dbt.type_string() }}) as cash_flow_type,
+       ordinal
+   from {{ var('cash_flow_statement_type_ordinal') }}
+),
+{% endif %}
+ 
+cash_flow_types as (
+
+   select cash_flow_key.*,
+   {% if var('cash_flow_statement_type_ordinal') %}
+       coalesce(account_number_ordinal.cash_flow_type, account_sub_type_ordinal.cash_flow_type, account_type_ordinal.cash_flow_type, account_class_ordinal.cash_flow_type) as cash_flow_type 
+   {% else %}
+       case when account_type = 'Bank' then 'Cash or Cash Equivalents'
+           when account_type = 'Accounts Receivable' then 'Operating'
+           when account_type = 'Credit Card' then 'Operating'
+           when account_type = 'Other Current Asset' then 'Operating'
+           when account_type = 'Accounts Payable' then 'Operating'
+           when account_type = 'Other Current Liability' then 'Operating'
+           when account_name = 'Net Income Adjustment' then 'Operating'
+           when account_type = 'Fixed Asset' then 'Investing'
+           when account_type = 'Other Asset' then 'Investing'
+           when account_type = 'Long Term Liability' then 'Financing'
+           when account_class = 'Equity' then 'Financing'
+           end as cash_flow_type
+   {% endif %}
+   from cash_flow_key
+ 
+   {% if var('cash_flow_statement_type_ordinal') %}
+
+   {% set cash_flow_type_fields = ['account_number', 'account_sub_type', 'account_type', 'account_class'] %}
+ 
+   {% for cash_flow_type_field in cash_flow_type_fields %}       
+       left join ordinals as {{ cash_flow_type_field }}_ordinal
+           on cash_flow_key.{{ cash_flow_type_field }} = {{ cash_flow_type_field }}_ordinal.{{ cash_flow_type_field }}
+   {% endfor %}
+ 
+   {% endif %}
+),
+ 
+cash_flow_ordinals as (
+ 
+   select cash_flow_types.*,
+
+   {% if var('cash_flow_statement_type_ordinal') %}
+       coalesce(account_number_ordinal.ordinal, account_sub_type_ordinal.ordinal, account_type_ordinal.ordinal, account_class_ordinal.ordinal, cash_flow_type_ordinal.ordinal) as ordinal
+   {% else %}
+       case when cash_flow_type = 'Cash or Cash Equivalents' then 1
+           when cash_flow_type = 'Operating' then 2
+           when cash_flow_type  = 'Investing' then 3
+           when cash_flow_type  = 'Financing' then 4
+       end as ordinal
+   {% endif %}
+
+   from cash_flow_types
+
+   {% if var('cash_flow_statement_type_ordinal') %}
+ 
+   {% set ordinal_fields = ['cash_flow_type', 'account_number', 'account_sub_type', 'account_type', 'account_class'] %}
+ 
+   {% for ordinal_field in ordinal_fields %} 
+       left join ordinals as {{ ordinal_field }}_ordinal
+           on cash_flow_types.{{ ordinal_field }} = {{ ordinal_field }}_ordinal.{{ ordinal_field }}
```

Comment:
Why are we doing this join twice? Since we are doing it in the above cte, shouldn't we be able to combine the ordinal and cash flow type operations?

Reply:
I wonder if it is this join, or the other - but I feel they are causing an unexpected fanout when using the seed file to generate the ordinals. Would you be able to investigate and adjust the query to resolve the fanout when using the seed file.

Reply:
You were correct, I brought the `cash_flow_type` into the loop unnecessarily. Should be consolidated now.